testing/simapp/app.go

```diff
@@ -672,9 +672,9 @@ func (app *SimApp) LoadHeight(height int64) error {
 func (app *SimApp) ModuleAccountAddrs() map[string]bool {
 	modAccAddrs := make(map[string]bool)
 	for acc := range maccPerms {
-		// do not add mock module to blocked addresses
+		// do not add the following modules to blocked addresses
 		// this is only used for testing
-		if acc == ibcmock.ModuleName {
+		if acc == ibcmock.ModuleName || acc == distrtypes.ModuleName {
```

Comment:
This is pretty hacky I just left it in for discussion. I'm guessing we don't want to unblock the distribution module?

We have the following [test case](https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/keeper/msg_server_test.go#L188). We ensure a non-blocked module account can escrow fees. 

When I try to use the mock module account we get an AuthKeeper error saying the account doesn't exist. I'm not sure why exactly. I've left the test case as is using the distribution module account for now. 

cc: @colin-axner

Reply:
I was looking at the test case "refund account is a module account" and trying to use `ibcmock.ModuleName` instead of `disttypes.ModuleName`:

```
"refund account is module account",
func() {
	msg.Signer = suite.chainA.GetSimApp().AccountKeeper.GetModuleAddress(disttypes.ModuleName).String()
	expPacketFee := types.NewPacketFee(fee, msg.Signer, nil)
	expFeesInEscrow = []types.PacketFee{expPacketFee}
},
true,
```

The account does not exist error is being returned from [here in escrow.go](https://github.com/cosmos/ibc-go/blob/main/modules/apps/29-fee/keeper/escrow.go#L24). And interestingly if we actually use [`AccountKeeper.GetModuleAccount()`](https://github.com/cosmos/cosmos-sdk/blob/v0.46.0/x/auth/keeper/keeper.go#L211) we will not encounter that error as it calls directly into `GetModuleAccountAndPermissions()` which [creates and sets the account in `x/auth`](https://github.com/cosmos/cosmos-sdk/blob/v0.46.0/x/auth/keeper/keeper.go#L201-L204).

This differs from [`GetModuleAddress`](https://github.com/cosmos/cosmos-sdk/blob/v0.46.0/x/auth/keeper/keeper.go#L165) which just checks the `permAddrs` map created on keeper initialisation. 

It's definitely pretty ambiguous as to what constitutes a blocked address and what doesn't. Without this logical OR added for `distrtypes.ModuleName` it is and will be in the `blockedAddrs` map in `x/bank`. However, its usage in `x/distribution` when for example, [withdrawing delegation rewards](https://github.com/cosmos/cosmos-sdk/blob/e39d84e06fafc33dcec85db12242e5f481ed0385/x/distribution/keeper/delegation.go#L172) uses `SendCoinsFromModuleToAccount` which only checks that the [recipientAddr](https://github.com/cosmos/cosmos-sdk/blob/e39d84e06fafc33dcec85db12242e5f481ed0385/x/bank/keeper/keeper.go#L322) is not a blocked address.

When I use the following in the `malleate()` func, we don't fail on the account not existing but instead fail with an "insufficient funds" error.

```
moduleAcc := suite.chainA.GetSimApp().AccountKeeper.GetModuleAccount(suite.chainA.GetContext(), ibcmock.ModuleName)
msg.Signer = moduleAcc.GetAddress().String()
expPacketFee := types.NewPacketFee(fee, msg.Signer, nil)
expFeesInEscrow = []types.PacketFee{expPacketFee}
```

It would be nice to get some more clarity around module accounts in general.

Reply:
If the blocked address logic gets removed from the SDK, what test case would we want to have? I'd say probably still wise to just use the mock module account since it may not make sense long term to use distr (as community pool might be removed from it)

Maybe we can add the mock module to the list of genAccs in `NewTestChainWithValSet`?

Also I think you can use `GetModuleAddress().String()`